File: src/main/java/com/thunderbear06/ai/task/tasks/MoveToEntityTask.java
package com.thunderbear06.ai.task.tasks;

import com.thunderbear06.entity.android.AndroidEntity;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.ai.navigation.PathNavigation;

public class MoveToEntityTask extends EntityBasedTask {
    private final double moveSpeed;
    private final PathNavigation navigation;

    public MoveToEntityTask(AndroidEntity android, double moveSpeed, LivingEntity entity) {
        super(android, entity);
        this.moveSpeed = moveSpeed;
        this.navigation = android.getNavigation();
    }

    @Override
    public String getName() {
        return "movingToEntity";
    }

    @Override
    public boolean shouldTick() {
        return super.shouldTick() && !isInRange(1.5D);
    }

    @Override
    public void firstTick() {
    }

    @Override
    public void tick() {
        super.tick();
        if (navigation.isDone()) navigation.moveTo(getTarget(), moveSpeed);
    }

    @Override
    public void lastTick() {
        navigation.stop();
    }
}

File: src/main/kotlin/com/thunderbear06/ai/task/tasks/MoveToEntityTask.kt
package com.thunderbear06.ai.task.tasks

import com.thunderbear06.entity.android.AndroidEntity
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.ai.navigation.PathNavigation

open class MoveToEntityTask(android: AndroidEntity, private val moveSpeed: Double, entity: LivingEntity) : EntityBasedTask(android, entity) {
    private val navigation: PathNavigation = android.navigation

    override val name: String = "movingToEntity"

    override fun shouldTick(): Boolean = super.shouldTick() && !isInRange(1.5)

    override fun firstTick() {
    }

    override fun tick() {
        super.tick()
        if (navigation.isDone) navigation.moveTo(target, moveSpeed)
    }

    override fun lastTick() {
        navigation.stop()
    }
}
